zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java

```diff
@@ -186,6 +187,44 @@ public ActorFuture<Void> resumeExporting() {
         });
   }
 
+  /**
+   * Disables an already configured exporter. No records will be exported to this exporter anymore.
+   *
+   * @param exporterId id of the exporter to disabled
+   * @return future which will be completed after the exporter is disabled.
+   */
+  public ActorFuture<Void> disableExporter(final String exporterId) {
+    if (actor.isClosed()) {
+      return CompletableActorFuture.completed(null);
+    }
+
+    return actor.call(() -> removeExporter(exporterId));
```

Comment:
:question: There's technically a race condition here, checking if the actor is closed and calling the actor (could be closed in between). So in one case, we "succeed" implicitly (if the actor is closed), but if it gets closed before call, we throw an exception. So callers should somehow handle the exception? Or is the expectation that we shouldn't call this if the actor is closed? :thinking: 

I imagine the impact will be quite low, but it's one of those things where it would be nice to have more control over (e.g. either `actor.callIfPossible`, or `try { actor.call } catch (ActorClosedException e) { // ignore }` :shrug:)

Reply:
I think it is ok to expect the caller to handle this exception. The actor gets closed usually during role transition. In worst case, the operation will be retried by the ConfigManager after a delay. So eventually the operation gets completed. But we can check if we can handle this differently in the caller - which will be either the `PartitionManger` or `ZeebePartition`.

Reply:
:thinking: What does it look like if I want to handle it as the caller? I know if I join, I get a generic `ExecutionException`, which is hard to distinguish from any other error. Is it easy to differentiate in general from other possible errors when used as a plain actor future?

Reply:
I thought the future fails with a specific "ActorClosedException", but it seems it us just a plain RuntimeException with message "Actor is closed". :smile: May be we should fix this in the Actor and fail the future with a specific exception.